### PR TITLE
Issue#1203 users should not be able to reset exams

### LINKED ISSUE #1203: Cuando reseteo un guia del tipo examen, tambien se resetan la cantidad de soluciones incorrectas enviadas en los ejercicios
O bien no deberia poder resetear una guia del tipo examen, o bien no deberia resetearse la cantidad de submissions incorrectas enviadas.

### PR DESCRIPTION
**Pull request by julian-berbel**
Fixes #1203.

### CODE REVIEW

**Comment by luchotc** [comment]
Nice one @julian-berbel! :clap:

File: app/controllers/guide_progress_controller.rb [hunk 1]
```diff
@@ -1,7 +1,18 @@
 class GuideProgressController < ApplicationController
+  before_action :check_not_in_exam!
```
Comment (luchotc) [inline app/controllers/guide_progress_controller.rb]: Maybe we should specify that this action is only for destroy, to make it more action revealing for the future.

Comment (julian-berbel) [reply app/controllers/guide_progress_controller.rb]: yeah, that's fair

File: app/controllers/guide_progress_controller.rb [hunk 2]
```diff
@@ -1,7 +1,18 @@
 class GuideProgressController < ApplicationController
+  before_action :check_not_in_exam!, only: :destroy
+
   def destroy
-    guide = Guide.find(params[:guide_id])
     guide.clear_progress! current_user
```
Comment (flbulgarelli) [inline app/controllers/guide_progress_controller.rb]: Shouldn't we change this to @guide?

Comment (julian-berbel) [reply app/controllers/guide_progress_controller.rb]: no?

Comment (flbulgarelli) [reply app/controllers/guide_progress_controller.rb]: Please ignore this comment. I did not see the corresponding method

**Pull request merged**